### PR TITLE
boards: arm: nrf: Make Bluetooth optional

### LINKED ISSUE #5454: [nRF] The nRF boards should not enable networking stacks
It is IMHO undesired behaviour that nrf52840_pca10056_defconfig and friends are enabling SW components.

Especially big SW components like BT.

HW Kconfig should be describing the available HW, not making assumptions about what SW is going to be used.

Networking stacks should be pulled in directly or indirectly because they are needed by applications.

### LINKED ISSUE #12215: Nordic: remove default-enable of bluetooth capability
As discussed in #12209 it's been discovered that (almost?) all Nordic-based boards have the following in their board defconfig:

```
# bluetooth
CONFIG_BT=y
CONFIG_BT_CTLR=y
```

This causes any application built for the boards to get roughly 30 KB of possibly unnecessary driver infrastructure.  It's particularly an issue when another transport mechanism (802.15.4 or OpenThreads) is desired instead.

Task is to review why this was done and, if there isn't a compelling explanation, remove this from all Nordic board defconfig files.

### PR DESCRIPTION
**Pull request by carlescufi**
Instead of enabling Bluetooth by default on nRF5x boards, only enable
the controller if Bluetooth has been enabled by the applicaiton.

Fixes #5454
Fixes #12215

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
Looks fine - I only hope that there are no apps that rely on the board doing CONFIG_BT=y and leave it out in their prj.conf.

**Comment by carlescufi** [comment]
>Looks fine - I only hope that there are no apps that rely on the board doing CONFIG_BT=y and leave it out in their prj.conf.

In-tree apps can be fixed easily. Those that are out-of-tree should of course enable it in `prj.conf`. If someone complains we can help out.

**Review by mbolivar — APPROVED** [review]

**Review by mike-scott — APPROVED** [review]
LGTM

**Review by pabigot — APPROVED** [review]

File: boards/arm/96b_carbon_nrf51/Kconfig.defconfig [hunk 1]
```diff
@@ -9,4 +9,11 @@ if BOARD_96B_CARBON_NRF51
 config BOARD
 	default "96b_carbon_nrf51"
 
+if BT
+
+config BT_CTLR
+	default y
+
+endif # BT
```
Comment (ulfalizer) [inline boards/arm/96b_carbon_nrf51/Kconfig.defconfig]: Could simplify these to make things a bit less spammy:

```
config BT_CTLR
	default BT
```

Comment (pabigot) [reply boards/arm/96b_carbon_nrf51/Kconfig.defconfig]: That's exactly the sort of cleaner solution I was hoping would get discussed around https://github.com/zephyrproject-rtos/zephyr/pull/12769#pullrequestreview-196824143.  A fourth way to do it....

Comment (carlescufi) [reply boards/arm/96b_carbon_nrf51/Kconfig.defconfig]: Ah, nice suggestion! let me do that!

**Comment by carlescufi** [comment]
@ulfalizer @pabigot using the new idiom now, thanks!

**Review by galak — COMMENTED** [review]
Hmm, now I'm wondering can we put this somewhere common and not have to do it in every board?

**Pull request merged**